media_test_harness/runtests.py

```diff
@@ -28,40 +72,72 @@ def __init__(self, **kwargs):
                         help='ini file of urls to make available to all tests')
 
     def parse_args(self, *args, **kwargs):
-        options, test_files = FirefoxUIOptions.parse_args(self,
-                                                          *args, **kwargs)
+        options, tests = BaseMarionetteOptions.parse_args(self, *args, **kwargs)
+
         if options.urls:
             if not os.path.isfile(options.urls):
                 self.error('--urls must provide a path to an ini file')
             else:
                 path = os.path.abspath(options.urls)
-                options.video_urls = self.get_urls(path)
+                options.video_urls = MediaTestOptions.get_urls(path)
         else:
             default = os.path.join(firefox_media_tests.urls, 'default.ini')
             options.video_urls = self.get_urls(default)
 
-        # replace firefox_ui_tests with firefox_media_tests
-        if set(test_files) <= {ui_manifest, puppeteer_manifest}:
-            test_files = [firefox_media_tests.manifest]
+        tests = tests or [firefox_media_tests.manifest]
 
-        return (options, test_files)
+        return options, tests
 
-    def get_urls(self, manifest):
+    @staticmethod
+    def get_urls(manifest):
         with open(manifest, 'r'):
             return [line[0] for line in read_ini(manifest)]
 
 
-class MediaTestRunner(FirefoxUITestRunner):
-    def __init__(self, *args, **kwargs):
-        FirefoxUITestRunner.__init__(self, *args, **kwargs)
-        if not self.server_root or self.server_root == ui_resources:
+class MediaTestRunner(BaseMarionetteTestRunner):
+    def __init__(self, **kwargs):
+        BaseMarionetteTestRunner.__init__(self, **kwargs)
+        if not self.server_root:
             self.server_root = firefox_media_tests.resources
+        self.prefs.update(DEFAULT_PREFS)
+
+        if not kwargs.get('e10s'):
+            self.prefs.update({'browser.tabs.remote.autostart': False})
+
         self.test_handlers = [MediaTestCase]
```

Comment:
Oh, here is where you are setting e10s. Does this happen before FF starts up? Because FF has to rebooted after changing this setting.